_layouts/default.html

```diff
@@ -26,11 +26,13 @@ <h2>{{ page.description | default: site.description | default: site.github.proje
           <a href="{{ site.github.zip_url }}" id="download-zip" class="button"><span>Download .zip</span></a>
           <a href="{{ site.github.tar_url }}" id="download-tar-gz" class="button"><span>Download .tar.gz</span></a>
           {% endif %}
+	{% unless site.github.hide_view_on_github %}
```

Comment:
What's the advantage of using an explicit setting versus `site.github.public` (which is set by the build process)? When would a user not want the banner on a public repository (or want it on a private one)?